Source/carthage/Archive.swift

```diff
@@ -71,7 +71,7 @@ public struct ArchiveCommand: CommandType {
 					}
 				}
 				.collect()
-				.map { Array(Set($0)).sort() }
+				.map { Array(Set($0.filter{ $0.hasSuffix(".framework") })).sort() }
```

Comment:
> There's a private `shouldBuildFrameworkType` function in `Xcode.swift` that we could make `internal`. 

Since that is in `CarthageKit` module, and the code here is in `carthage` module, it should be `public` to reuse it.